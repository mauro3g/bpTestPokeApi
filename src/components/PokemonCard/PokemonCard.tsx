import React from 'react'
import { IPokemon } from '../../types/types'

const style: { [key: string]: React.CSSProperties } = {
    card: {
        color: 'white',
        textAlign: 'center',
        fontSize: '2rem',
        padding: '10px 10px',
        backgroundColor: '#424242',
        width: '21vw',
        height: '18vh',
        display: 'flex',
        alignItems: 'center',
        margin: '10px'
    },
    image: {
        backgroundColor: 'white',
        width: '70px',
        height: '70px',
        borderRadius: '50%',
        marginLeft: '10px'
    },
    textContainer: {
        flexGrow: '1',
        color: 'white',
        height: '90%',
        marginLeft: '15px',
        fontSize: '1rem'
    },
    statsBox: {
        display: 'flex',
        justifyContent: 'space-between',
        padding: '0px'
    },
    actions: {
        height: '80%',
        display: 'flex',
        justifyContent: 'space-around',
        flexDirection: 'column',
    },
    actionButton: {
        fontSize: '1.2rem',
        margin: '0px 10px',
        backgroundColor: 'white',
        borderRadius: '50%',
        width: '35px',
        height: '35px'
    }
} as const

interface Props {
    pokemon: IPokemon
    index: number
    handleOpenForm: (index: number) => void,
    handleDeletePokemon: (pokemonId: number) => Promise<void>
}

const PokemonCard = (props: Props) => {
    const { pokemon, index, handleOpenForm, handleDeletePokemon } = props

    const handleDelete = () => {
        handleDeletePokemon(pokemon.id as number)
    }

    return (
        <div style={style.card} data-testid="card-component">
            <div >
                <img style={style.image} src={pokemon.image} alt='pokemon'></img>
            </div>
            <div style={style.textContainer}>
                <p>{pokemon.name}</p>
                <div>{pokemon.type}</div>
                <div style={style.statsBox}>
                    <p>Atk: {pokemon.attack} </p>
                    <p>Def: {pokemon.defense} </p>
                    <p>Hp: {pokemon.hp} </p>
                </div>
            </div>
            <div style={style.actions}>
                <button style={style.actionButton} type="button" onClick={(e) => handleOpenForm(index)}><i className="fa fa-pencil"></i></button>
                <button style={style.actionButton} type="button" onClick={(e) => handleDelete()}><i className="fa fa-times-circle-o"></i></button>
            </div>
        </div>
    )
}

export default PokemonCard