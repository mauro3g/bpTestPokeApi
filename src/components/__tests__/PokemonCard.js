import { render, screen, cleanup } from '@testing-library/react';
import PokemonCard from '../PokemonCard/PokemonCard'
import renderer from 'react-test-renderer'

const defaultPokemon = {
    id: 0,
    name: 'Pikachu',
    image: "https://assets.pokemon.com/assets/cms2/img/pokedex/full/012.png",
    type: 'fire',
    hp: 50,
    attack: 50,
    defense: 50,
    idAuthor: 1
}


test('should render a button component', () => {
    render(<PokemonCard pokemon={defaultPokemon} index={0} handleOpenForm={()=>{}} handleDeletePokemon={()=>{}}></PokemonCard>)
    const element = screen.getByTestId('card-component')
    expect(element).toBeInTheDocument()
    expect(element).toHaveTextContent(defaultPokemon.name)
})

test('match snapshop', ()=>{
    const testBtn = renderer.create(<PokemonCard pokemon={defaultPokemon} index={0} handleOpenForm={()=>{}} handleDeletePokemon={()=>{}}></PokemonCard>).toJSON()
    expect(testBtn).toMatchSnapshot()
})

