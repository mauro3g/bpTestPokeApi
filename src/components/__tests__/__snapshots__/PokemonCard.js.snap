// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshop 1`] = `
<div
  data-testid="card-component"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#424242",
      "color": "white",
      "display": "flex",
      "fontSize": "2rem",
      "height": "18vh",
      "margin": "10px",
      "padding": "10px 10px",
      "textAlign": "center",
      "width": "21vw",
    }
  }
>
  <div>
    <img
      alt="pokemon"
      src="https://assets.pokemon.com/assets/cms2/img/pokedex/full/012.png"
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": "50%",
          "height": "70px",
          "marginLeft": "10px",
          "width": "70px",
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "color": "white",
        "flexGrow": "1",
        "fontSize": "1rem",
        "height": "90%",
        "marginLeft": "15px",
      }
    }
  >
    <p>
      Pikachu
    </p>
    <div>
      fire
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
          "padding": "0px",
        }
      }
    >
      <p>
        Atk: 
        50
         
      </p>
      <p>
        Def: 
        50
         
      </p>
      <p>
        Hp: 
        50
         
      </p>
    </div>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "height": "80%",
        "justifyContent": "space-around",
      }
    }
  >
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": "50%",
          "fontSize": "1.2rem",
          "height": "35px",
          "margin": "0px 10px",
          "width": "35px",
        }
      }
      type="button"
    >
      <i
        className="fa fa-pencil"
      />
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": "50%",
          "fontSize": "1.2rem",
          "height": "35px",
          "margin": "0px 10px",
          "width": "35px",
        }
      }
      type="button"
    >
      <i
        className="fa fa-times-circle-o"
      />
    </button>
  </div>
</div>
`;
